http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -221,6 +263,124 @@ public void accept(ChannelHandlerContext ctx, io.netty.handler.codec.http.HttpRe
+            return ExecutionLeaf.indeterminate();
+        }
+        boolean unwrapResponse = HttpResponse.class.isAssignableFrom(executionHandle.getReturnType().getType());
+        MatchRule.ContentType fixedContentType = findFixedContentType(rule);
+        MediaType responseMediaType;
+        if (fixedContentType != null) {
+            responseMediaType = fixedContentType.expectedType();
+        } else {
+            List<MediaType> produces = routeInfo.getProduces();
+            if (!produces.isEmpty()) {
+                responseMediaType = produces.get(0);
+            } else {
+                responseMediaType = MediaType.APPLICATION_JSON_TYPE;
+            }
+        }
+        RequestArgumentBinder<Object>[] argumentBinders = routeInfo.resolveArgumentBinders(requestArgumentSatisfier.getBinderRegistry());
+        ShortCircuitArgumentBinder.Prepared[] shortCircuitBinders = new ShortCircuitArgumentBinder.Prepared[argumentBinders.length];
+        for (int i = 0; i < argumentBinders.length; i++) {
+            if (!(argumentBinders[i] instanceof ShortCircuitArgumentBinder<Object> scb)) {
+                return ExecutionLeaf.indeterminate();
+            }
+            //noinspection unchecked
+            Optional<ShortCircuitArgumentBinder.Prepared> prep = scb.prepare(executionHandle.getArguments()[i], fixedContentType);
+            if (prep.isEmpty()) {
+                return ExecutionLeaf.indeterminate();
+            }
+            shortCircuitBinders[i] = prep.get();
+        }
+        if (routeInfo.getExecutor(serverConfiguration.getThreadSelection()) != null ||
+            routeInfo.isSuspended() ||
+            routeInfo.isAsyncOrReactive()) {
+            return ExecutionLeaf.indeterminate();
+        }
+        if (!(executionHandle instanceof UnsafeExecutionHandle<?, ?> unsafeExecutionHandle)) {
+            return ExecutionLeaf.indeterminate();
+        }
+        @SuppressWarnings("unchecked")
+        MessageBodyWriter<Object> messageBodyWriter = (MessageBodyWriter<Object>) routeInfo.getMessageBodyWriter();
+        ShortCircuitNettyBodyWriter<Object> scWriter ;
+        RawMessageBodyHandler<Object> rawWriter ;
+        if (messageBodyWriter instanceof ShortCircuitNettyBodyWriter<Object> scw) {
+            rawWriter = null;
+            scWriter = scw;
+        } else if (messageBodyWriter instanceof RawMessageBodyHandler<Object> raw) {
+            rawWriter = raw;
+            scWriter = null;
+        } else {
+            return ExecutionLeaf.indeterminate();
+        }
+        return new ExecutionLeaf.Route<>(new RequestHandler() {
+            @Override
+            public void accept(ChannelHandlerContext ctx, io.netty.handler.codec.http.HttpRequest request, ByteBody body, PipeliningServerHandler.OutboundAccess outboundAccess) {
+                try {
+                    NettyHttpHeaders requestHeaders = new NettyHttpHeaders(request.headers(), conversionService);
```

Comment:
like I said do we need this headers object?

Reply:
see above